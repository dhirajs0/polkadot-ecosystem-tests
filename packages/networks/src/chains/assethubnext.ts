import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNext: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNext) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNext = defineChain({
  name: 'assetHubNext',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNext,
  initStorages: getInitStorages(custom.assetHubNext),
})
