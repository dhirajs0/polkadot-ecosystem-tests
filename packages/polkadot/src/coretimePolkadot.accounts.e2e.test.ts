import { coretimePolkadot } from '@e2e-test/networks/chains'
import { registerTestTree, transferFunctionsTests } from '@e2e-test/shared'

registerTestTree(
  transferFunctionsTests(coretimePolkadot, {
    testSuiteName: 'Polkadot Coretime Accounts',
    addressEncoding: 0,
    chainEd: 'Normal',
  }),
)
