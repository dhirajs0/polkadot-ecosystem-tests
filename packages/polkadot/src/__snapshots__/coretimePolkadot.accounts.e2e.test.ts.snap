// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Polkadot Coretime Accounts > \`transfer_allow_death\` > should allow killing sender account > events when Alice \`transfer_allow_death\` to Bob 1`] = `
[
  {
    "data": {
      "amount": "(rounded 22000000)",
      "who": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
    },
    "method": "Withdraw",
    "section": "balances",
  },
]
`;
